src/main/kotlin/calculator/domain/Calculator.kt

```diff
@@ -3,25 +3,18 @@ package calculator.domain
 class Calculator {
 
     fun calculate(numbers: List<String>): Int {
+        val numbers = convertStringToNumber(numbers)
         var sum = 0
         for (number in numbers) {
-            checkValidateNumber(number)
-            checkNegativeNumber(number)
-            sum += number.toInt()
+            sum += number.value
         }
 
         return sum
     }
 
-    private fun checkValidateNumber(string: String) {
-        try {
-            string.toInt()
-        } catch (e: NumberFormatException) {
-            throw RuntimeException("$string 은 올바른 숫자가 아닙니다.")
+    fun convertStringToNumber(numbers: List<String>): List<Number> {
+        return numbers.map { stringNumber ->
+            Number(stringNumber)
         }
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        return numbers.map(::Number)
```

Reply:
이 기능은 뭐라고 부르나요??